src/routers/router.ts

```diff
@@ -0,0 +1,34 @@
+import { Router } from 'express';
+import config from '../config';
+import { IGroup, User } from '../group/group.interface';
+import GroupService from '../group/group.service';
+import UserService from '../user/user.service';
+import wrapAsync from '../utils/wrappers';
+
+const router = Router();
+
+// Get group by ID
+router.get('/groups/:id', wrapAsync(async (req, res) => {
+  const groupID = req.params.id;
+  const requesterID = req.headers[config.userHeader] as string;
+
+  const axiosRes = await GroupService.getByID(groupID, requesterID);
+  const group = axiosRes.data as IGroup;
+  const members = group.users;
+  let isPartialContent = false;
+
+  const populatedUsers = await Promise.all(members.map(async (member: User) => {
+    try {
+      const userAxiosRes = await UserService.getByID(member.id);
+      return { ...member, user: userAxiosRes.data };
+    } catch {
+      isPartialContent = true;
+      return member;
+    }
+  }));
+  res
+    .status(isPartialContent ? 200 : 206)
```

Comment:
I suggest to add an explations remer of the status
```suggestion
    .status(isPartialContent ? 200 : 206) //200: completely populated, 206: partialy populated 
```

Reply:
Why do you think that comment is necessary here?

Reply:
because when I read it, it took time to understand 
There are a lot of statuses in use

Reply:
I don't think we have to comment over HTTP status codes.
Status `200 OK` is probably the most known one.
Indeed, status `206 Partial Content` is less known, but I think  the `isPartialContent` explains its purpose clearly enough.